packages/build/tests/blobs_upload/tests.js

```diff
@@ -222,3 +245,55 @@ if (semver.gte(nodeVersion, '18.19.0')) {
     t.deepEqual(blob3.metadata, { some: 'metadata' })
   })
 }
+
+test.serial('Blobs upload failure print full error stack and cause to systemlog', async (t) => {
+  const fixture = await new Fixture('./fixtures/src_with_blobs').withCopyRoot({ git: false })
+
+  const systemLogFile = await tmp.file()
+
+  const {
+    success,
+    logs: { stdout, stderr },
+  } = await fetchCustomImplementationStore.run(
+    {
+      fetchImplementation: (origFetch, ...args) => {
+        if (
+          typeof args[0] === 'string' &&
+          args[0].includes('api/v1/blobs') &&
+          typeof args[1] === 'object' &&
+          args[1].method === 'put'
+        ) {
+          throw new Error('Simulated upload error with cause', {
+            cause: new Error('Outer internal error', { cause: new Error('Nested internal error') }),
+          })
```

Comment:
This is somewhat simulating fetch error that often will have `cause` - for example like so:
```
TypeError: fetch failed
    at Object.fetch (node:internal/deps/undici/undici:11118:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: Error: connect ECONNREFUSED 127.0.0.1:1337
      at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1300:16) {
    errno: -111,
    code: 'ECONNREFUSED',
    syscall: 'connect',
    address: '127.0.0.1',
    port: 1337
  }
}
```
As `fetch failed` is just category of error that can have multiple underlying causes